source/simulator/lateral_averaging.cc

```diff
@@ -149,9 +149,9 @@ namespace aspect
           for (unsigned int q=0; q<output.size(); ++q)
             {
               const Tensor<1,dim> g = gravity_->gravity_vector(in.position[q]);
-              const Tensor<1,dim> vertical = (g.norm() > 0 ? g/g.norm() : Tensor<1,dim>());
+              const Tensor<1,dim> vertically_down = (g.norm() > 0 ? g/g.norm() : Tensor<1,dim>());
 
-              output[q] = std::fabs(std::min(0.0, velocity_values[q] * vertical))
+              output[q] = std::max(0.0, velocity_values[q] * vertically_down)
```

Comment:
Oh, had this always been wrong!?!?

Reply:
I think so. The last changes to that file were 6 years ago...
I think this was just hard to notice, because sinking and rising velocities are so similar when all boundaries are closed. I just noticed it because I computed the actual rising velocity, and one of the test cases was a pipe flow model.